dotCMS/src/main/java/com/dotmarketing/fixtask/tasks/FixTask00095DeleteOrphanRelationships.java

```diff
@@ -0,0 +1,200 @@
+                    + fixesUriSubstring).exists()) {
+                new File(ConfigUtils.getBackupPath() + File.separator + fixesUriSubstring)
+                        .mkdirs();
+            }
+            writingObj = new File(ConfigUtils.getBackupPath() + File.separator
+                    + fixesUriSubstring + File.separator + lastModDate + "_"
+                    + "FixTask00095DeleteOrphanRelationships" + ".xml");
+
+            BufferedOutputStream bufferedOutObj = null;
+            try {
+                bufferedOutObj = new BufferedOutputStream(new FileOutputStream(writingObj));
+            } catch (FileNotFoundException e) {
+                Logger.error(FixTask00095DeleteOrphanRelationships.class, 
+                        "Could not write to Fix Task status file.", e);
+            }
+            xstreamObj.toXML(new ArrayList<>(modifiedData), bufferedOutObj);
+        }
+        return new ArrayList<>(modifiedData);
+    }
+
+    @Override
+    public boolean shouldRun() {
+        Logger.debug(CMSMaintenanceFactory.class, "Running query for relationships: "
+                + VERIFICATION_QUERY);
+        DotConnect dc = new DotConnect();
+        dc.setSQL(VERIFICATION_QUERY);
+        List<Map<String, Object>> inodesInStructure = new ArrayList<>();
+        
+        try {
+            inodesInStructure = dc.loadObjectResults();
+        } catch (DotDataException e) {
+            Logger.error(this, e.getMessage(), e);
+        }
+        
+        final int total = (inodesInStructure != null && !inodesInStructure.isEmpty()) ? 
+                inodesInStructure.size() : 0;
+                
+        Logger.debug(CMSMaintenanceFactory.class, "Found " + total + " invalid inodes.");
+        
+        FixAssetsProcessStatus.setTotal(total);
+        return total > 0 ? true : false;
+    }
+
+    
+    private void startCleanup() throws DotHibernateException {
+        try {
+            HibernateUtil.startTransaction();
+            cleanUpOrphanRelationships();
+            HibernateUtil.commitTransaction();
+            // Set the number of records that were fixed
+            FixAssetsProcessStatus.setErrorsFixed(modifiedData.size());
+        } catch (DotHibernateException e1) {
+            Logger.error(this, "Unable to clean orphaned relationships.", e1);
+            HibernateUtil.rollbackTransaction();
+            modifiedData.clear();
+        } catch (SQLException e2) {
+            Logger.error(this, "There was an unexpected problem with cleaning up relationship in Database.", e2);
+            HibernateUtil.rollbackTransaction();
+            modifiedData.clear();
+        } finally {
+            flushCacheRegions();
+        } 
+    }
+    
+    private void saveFixAuditRecord(final int total) throws DotHibernateException {
+        final FixAudit auditObj = new FixAudit();
+        auditObj.setTableName("relationship");
+        auditObj.setDatetime(new Date());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use the Java 8 Date and Time API instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2143)